### PR TITLE
Fixes for false positives in unused-components rule

### PR DESCRIPTION
**Pull request by lobocv**
- Recursively build reference maps in unused-components check

### CODE REVIEW

**Comment by lobocv** [comment]
This relies on https://github.com/pb33f/libopenapi/pull/322 which is why the build is failing

**Comment by daveshanley** [comment]
https://github.com/pb33f/libopenapi/pull/322 was merged.

File: functions/openapi/unused_component_test.go [hunk 1]
```diff
@@ -46,22 +51,7 @@ components:
...
-
-	rule := buildOpenApiTestRuleAction(path, "unused_component", "", nil)
-	ctx := buildOpenApiTestContext(model.CastToRuleAction(rule.Then), nil)
-	config := index.CreateOpenAPIIndexConfig()
-	ctx.Index = index.NewSpecIndexWithConfig(&rootNode, config)
-
-	def := UnusedComponent{}
-	res := def.RunRule(nodes, ctx)
-
+	res := setupUnusedComponentsTestContext(t, yml)
```
Comment (lobocv) [inline functions/openapi/unused_component_test.go:54]: I removed a bunch of copy-paste from each test

File: motor/rule_applicator.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ type ruleContext struct {
 // of ApplyRulesToRuleSet to change, without a huge refactor. The ApplyRulesToRuleSet function only returns a single error also.
 type RuleSetExecution struct {
 	RuleSet           *rulesets.RuleSet             // The RuleSet in which to apply
-	SpecFileName      string                        // The name of the specification file, used to correctly label location
+	SpecFileName      string                        // The path of the specification file, used to correctly label location
```
Comment (lobocv) [inline motor/rule_applicator.go:57]: Originally I had renamed this to SpecFilePath to be more clear it's the path, not just the name, but I realized that's a breaking change. So I just updated the docstring instead.

**Comment by lobocv** [comment]
@daveshanley Can we get this merged so it doesn't get stale? :)

**Review by daveshanley — APPROVED** [review]
LGTM!

**Pull request merged**